lib/utils.js

```diff
@@ -148,3 +158,59 @@ export async function storeError(errorMsg) {
 
   await update(queryError);
 }
+
+export async function updateStatusJob(jobUri, status) {
+  let dateModified = new Date();
+  const updateQuery = `
+    ${PREFIXES}
+    DELETE {
+      GRAPH ?g {
+        ?job adms:status ?status;
+           dct:modified ?modified.
+      }
+    }
+    WHERE {
+      GRAPH ?g {
+       BIND(${sparqlEscapeUri(jobUri)} AS ?job)
+       ?job a ${sparqlEscapeUri(JOB_TYPE)};
+         adms:status ?status;
+         dct:modified ?modified.
+      }
+    }
+    ;
+    INSERT DATA {
+      GRAPH ${sparqlEscapeUri(JOBS_GRAPH)}{
+        ${sparqlEscapeUri(jobUri)} adms:status ${sparqlEscapeUri(status)};
+           dct:modified ${sparqlEscapeDateTime(dateModified)}.
+      }
+    }
+  `;
+
+  await update(updateQuery);
+
+
+}
+
+export async function updateAndFilterTimedOutJobs(jobs) {
+  if (jobs.length && JOB_TIMEOUT_MINUTES) {
+    let timeoutMinutes = parseInt(JOB_TIMEOUT_MINUTES);
+    let stillActiveJobs = [];
+    for (const job of jobs) {
+      if (job.status === STATUS_BUSY || job.status === STATUS_SCHEDULED) {
+        let date = new Date(job.modified);
+        let now = new Date();
+
+        let diff = (now.getTime() - date.getTime()) * 1000 * 60;
+        if (diff >= timeoutMinutes) {
+          console.log(`job ${job.jobUri} has timed out. Setting status to failed...`);
+          await updateStatusJob(job.jobUri, STATUS_FAILED);
```

Comment:
`STATUS_FAILED` is not imported in the file

Reply:
oops, thank you 🙏🏼